modules/common/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/common/Config.scala

```diff
@@ -0,0 +1,160 @@
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.common
+
+import java.net.URI
+import java.util.UUID
+
+import cats.syntax.either._
+import cats.syntax.show._
+
+import io.circe._
+import io.circe.syntax._
+import io.circe.generic.semiauto._
+
+import com.snowplowanalytics.iglu.core.SchemaCriterion
+
+import com.typesafe.config.ConfigFactory
+
+import pureconfig._
+import pureconfig.module.circe._
+import pureconfig.error.{CannotParse, ConfigReaderFailures}
+
+import monocle.Lens
+import monocle.macros.GenLens
+
+import Config._
+
+/**
+ * Main config file parsed from HOCON
+ * @tparam D kind of supported warehouse
+ */
+final case class Config[+D <: StorageTarget](name: String,
+                                             id: UUID,
+                                             region: String,
+                                             jsonpaths: Option[S3.Folder],
+                                             monitoring: Monitoring,
+                                             messageQueue: String,
+                                             shredder: Shredder,
+                                             storage: D,
+                                             formats: Formats,
+                                             steps: Set[Step])
+
+object Config {
+
+  def fromString(s: String): Either[String, Config[StorageTarget]] =
+    Either
+      .catchNonFatal(ConfigSource.fromConfig(ConfigFactory.parseString(s)))
+      .leftMap(error => ConfigReaderFailures(CannotParse(s"Not valid HOCON. ${error.getMessage}", None)))
+      .flatMap { config =>
+        config
+          .load[Json]
+          .flatMap { json =>
+            json.as[Config[StorageTarget]].leftMap(failure => ConfigReaderFailures(CannotParse(failure.show, None)))
+          }
+      }
+      .leftMap(_.prettyPrint())
+      .flatMap { config =>
+        val overlaps = config.formats.findOverlaps
+        val message =
+          s"Following schema criterions overlap in different groups (TSV, JSON, skip): " +
+            s"${overlaps.map(_.asString).mkString(", ")}. " +
+            s"Make sure every schema can have only one format"
+        Either.cond(overlaps.isEmpty, config, message)
+      }
+
+  final case class Shredder(input: URI, output: URI, outputBad: URI, compression: Compression)
```

Comment:
Is there a reason why not making them `Folder` instead of `URI` ?

Reply:
Yes! They were `S3.Folder` at first, but then we need to have this config in tests and tests use local FS.

Reply:
I see !